lib/actions/collection.js

```diff
@@ -9,24 +9,30 @@ module.exports = {
       postFetchAction,
       preFetchAction,
       trackKey,
+      adapter = defaultAdapter,
     } = options;
 
-    const storageManager = StorageManager.get();
-
     if (xhrs[trackKey] && xhrs[trackKey].state() === 'pending') xhrs[trackKey].abort();
 
-    return (dispatch) => {
+    return (dispatch, getState) => {
       if (preFetchAction) dispatch(preFetchAction);
 
-      const xhr = storageManager.storage(brainstemKey)
-        .fetch(Object.assign(fetchOptions, { remove: false }));
+      const xhr = adapter.fetchCollection(brainstemKey, {
+        dispatch,
+        getState,
+        fetchOptions,
+      });
 
-      if (postFetchAction) xhr.done(collection => dispatch(postFetchAction(collection.pluck('id'))));
+      if (postFetchAction) {
+        xhr.done(collection =>
```

Comment:
How is `.done` going to work when the XHR is not from jQuery?

Reply:
We have our mavenlink adapter return a deferred object like so: 

```
fetchCollection: (brainstemKey, { dispatch, fetchOptions }) => {
       const deferred = $.Deferred();
 
       networkHelpers.fetchFromAPI(baseUrl, { params: fetchOptions.params, method: 'GET' })
         .then((result) => {
           dispatch(brainstemActions.syncCollections(result));
           deferred.resolve(result[brainstemKey]);
         });
 
       return deferred;
     },
```